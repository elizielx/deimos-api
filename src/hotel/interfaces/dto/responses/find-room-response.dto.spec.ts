import { FindRoomsResponseDTO } from "./find-room-response.dto";

describe("FindRoomsResponseDTO", () => {
    it("should create a new instance of FindRoomsResponseDTO", () => {
        const rooms = [
            {
                id: "1",
                number: 101,
                type: "single",
                rate: 100,
                status: "available",
            },
            {
                id: "2",
                number: 102,
                type: "double",
                rate: 150,
                status: "booked",
            },
        ];

        const findRoomsResponseDTO = new FindRoomsResponseDTO(rooms);

        expect(findRoomsResponseDTO.rooms).toEqual(rooms);
    });

    it("matches the snapshot", () => {
        const rooms = [
            {
                id: "1",
                number: 101,
                type: "single",
                rate: 100,
                status: "available",
            },
            {
                id: "2",
                number: 102,
                type: "double",
                rate: 150,
                status: "booked",
            },
        ];

        const findRoomsResponseDTO = new FindRoomsResponseDTO(rooms);

        expect(findRoomsResponseDTO).toMatchSnapshot();
    });
});
