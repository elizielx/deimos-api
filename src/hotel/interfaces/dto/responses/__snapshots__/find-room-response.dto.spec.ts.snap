// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindRoomsResponseDTO matches the snapshot 1`] = `
FindRoomsResponseDTO {
  "rooms": [
    {
      "id": "1",
      "number": 101,
      "rate": 100,
      "status": "available",
      "type": "single",
    },
    {
      "id": "2",
      "number": 102,
      "rate": 150,
      "status": "booked",
      "type": "double",
    },
  ],
}
`;
